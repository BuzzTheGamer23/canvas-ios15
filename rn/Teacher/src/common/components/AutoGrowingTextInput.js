// @flow

import React, { PureComponent } from 'react'
import { TextInput } from 'react-native'

type State = { height: number }
type Props = {
  defaultHeight: number,
  style?: any,
  onContentSizeChange?: Function,
  extraHeight: number,
}

export default class AutoGrowingTextInput extends PureComponent {
  props: Props
  state: State

  static defaultProps = {
    extraHeight: 0,
  }

  constructor (props: Props) {
    super(props)

    this.state = { height: this.props.defaultHeight }
  }

  updateContentSize = (e: any) => {
    this.setState({ height: Math.max(e.nativeEvent.contentSize.height + this.props.extraHeight, this.props.defaultHeight) })
    this.props.onContentSizeChange && this.props.onContentSizeChange(e)
  }

  render () {
    return (
      <TextInput
        {...this.props}
        style={[this.props.style, { height: this.state.height }]}
        multiline
        onContentSizeChange={this.updateContentSize}
      />
    )
  }
}
