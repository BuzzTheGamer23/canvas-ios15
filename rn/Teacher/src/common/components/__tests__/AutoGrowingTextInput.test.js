// @flow

import React from 'react'
import AutoGrowingTextInput from '../AutoGrowingTextInput'
import renderer from 'react-test-renderer'

let defaultProps = {
  defaultHeight: 54,
  onContentSizeChange: jest.fn(),
}

describe('AutoGrowingTextInput', () => {
  beforeEach(() => jest.resetAllMocks())
  it('renders', () => {
    let tree = renderer.create(
      <AutoGrowingTextInput {...defaultProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('adjusts the height when content size is updated', () => {
    let tree = renderer.create(
      <AutoGrowingTextInput {...defaultProps} />
    )

    let event = { nativeEvent: { contentSize: { height: 100 } } }

    tree.getInstance().updateContentSize(event)
    expect(tree.toJSON()).toMatchSnapshot()
    expect(defaultProps.onContentSizeChange).toHaveBeenCalledWith(event)
  })
})
