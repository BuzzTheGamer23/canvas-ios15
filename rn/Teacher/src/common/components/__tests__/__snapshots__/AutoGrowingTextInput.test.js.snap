// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoGrowingTextInput adjusts the height when content size is updated 1`] = `
<TextInput
  defaultHeight={54}
  extraHeight={0}
  multiline={true}
  onContentSizeChange={[Function]}
  style={
    Array [
      undefined,
      Object {
        "height": 100,
      },
    ]
  }
/>
`;

exports[`AutoGrowingTextInput renders 1`] = `
<TextInput
  defaultHeight={54}
  extraHeight={0}
  multiline={true}
  onContentSizeChange={[Function]}
  style={
    Array [
      undefined,
      Object {
        "height": 54,
      },
    ]
  }
/>
`;
